core-common/src/main/java/org/zalando/nakadi/repository/kafka/AvroDeserializerWithSequenceDecoder.java

```diff
@@ -39,8 +39,11 @@ public byte[] deserializeAvro(final EnvelopeHolder envelope) throws RuntimeExcep
             metadata.put(NakadiAvroMetadata.OCCURRED_AT, new DateTime(
                     (long) metadata.get(NakadiAvroMetadata.OCCURRED_AT), DateTimeZone.UTC).toString());
 
-            metadata.put(NakadiAvroMetadata.RECEIVED_AT, new DateTime(
-                    (long) metadata.get(NakadiAvroMetadata.RECEIVED_AT), DateTimeZone.UTC).toString());
+            final var receivedAt = metadata.get(NakadiAvroMetadata.RECEIVED_AT);
+            if (receivedAt != null) {
```

Comment:
why that could be the case ?

Reply:
The `received_at` is can be null according to the schema. So if the the enrichment fail for some reason, the following line will throw `NullpointerException`. So I decided that it is better to deliver the event without `received_at` timestamp than blocking consumption creating a `NullpointerException` in Nakadi consumption stack.

Reply:
> So if the the enrichment fail for some reason

that means event publishing should fail as well.

Reply:
The publishing part of Nakadi is supposed to put `received_at` in the metadata. But, there might be a mistake configuration or a bug that result in Nakadi sending an event without `received_at` in the metadata in Kafka. Most likely it will never happen, but still the consuming side of Kafka should handle it without breaking because the schema of metadata allows it.